### PR TITLE
Upgrade to Hibernate 6.4

### PR DESCRIPTION
**Pull request by mraible**
Test to see if CI passes...

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/generator-constants.mjs [hunk 1]
```diff
@@ -41,7 +41,7 @@ export const javaDependencies = {
    * hibernate version should match the one managed by https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-dependencies/SPRING_BOOT_VERSION
    * Required due to hibernate-jpamodelgen annotation processor.
    */
-  hibernate: '6.3.1.Final',
+  hibernate: '6.4.0.Final',
```
Comment (mshima) [inline generators/generator-constants.mjs:44]: We use hibernate from spring-boot/jhipster-bom. IFAIK this version is only used for `hibernate-jpamodelgen` dependency.

**Comment by DanielFran** [comment]
Spring-boot 3.2.0 uses 6.3.1.Final
https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-dependencies/3.2.0

File: generators/generator-constants.mjs (hunk 1 above)
Comment (mraible) [reply generators/generator-constants.mjs:44]: Good to know. Thanks!

**Comment by mraible** [comment]
@DanielFran Yep, I was trying to see if we could upgrade beyond it. Since this PR doesn't really change the version, I'll close it.

**Pull request closed without merging**